database.go

```diff
@@ -0,0 +1,161 @@
+// createDB создает новую базу данных и таблицу scheduler с индексом по полю date.
+func createDB(dbFile string) (*sql.DB, error) {
+	db, err := sql.Open("sqlite3", dbFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Создание таблицы scheduler
+	sqlStmt := `
+    CREATE TABLE scheduler (
+        id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+        date TEXT,
+        title TEXT,
+        comment TEXT,
+        repeat TEXT
+    );
+    `
+	_, err = db.Exec(sqlStmt)
+	if err != nil {
+		db.Close() // Закрываем базу, если произошла ошибка
+		return nil, fmt.Errorf("ошибка создания таблицы: %w", err)
+	}
+
+	// Создание индекса по полю date
+	indexStmt := `
+    CREATE INDEX idx_scheduler_date ON scheduler(date);
+    `
+	_, err = db.Exec(indexStmt)
+	if err != nil {
+		db.Close() // Закрываем базу, если произошла ошибка
+		return nil, fmt.Errorf("ошибка создания индекса: %w", err)
+	}
+
+	return db, nil
+}
+
+// openDB открывает существующую базу данных.
+func openDB(dbFile string) (*sql.DB, error) {
+	db, err := sql.Open("sqlite3", dbFile)
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+// getTaskByID возвращает задачу по идентификатору
+func getTaskByID(id string) (Task, error) {
+	var task Task
+	err := db.QueryRow("SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?", id).
+		Scan(&task.ID, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+	if err != nil {
+		return Task{}, err
+	}
+	return task, nil
+}
+
+// getTasks получает задачи из базы данных
+func getTasks(search string, limit int) ([]Task, error) {
+	var rows *sql.Rows
+	var err error
+
+	if search == "" {
+		// Если параметр search отсутствует, возвращаем ближайшие задачи
+		rows, err = db.Query("SELECT id, date, title, comment, repeat FROM scheduler ORDER BY date LIMIT ?", limit)
+	} else {
+		// Проверяем, является ли search датой
+		parsedDate, dateErr := time.Parse("02.01.2006", search)
+		if dateErr == nil {
+			// Если это дата, выбираем задачи на эту дату
+			rows, err = db.Query("SELECT id, date, title, comment, repeat FROM scheduler WHERE date = ? LIMIT ?", parsedDate.Format("20060102"), limit)
+		} else {
+			// Если это строка, ищем в title и comment
+			likeSearch := "%" + search + "%"
+			rows, err = db.Query("SELECT id, date, title, comment, repeat FROM scheduler WHERE title LIKE ? OR comment LIKE ? ORDER BY date LIMIT ?", likeSearch, likeSearch, limit)
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var tasks []Task
+	for rows.Next() {
+		var task Task
+		err := rows.Scan(&task.ID, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+		if err != nil {
+			return nil, err
+		}
+		tasks = append(tasks, task)
+	}
+
+	return tasks, nil
+}
+
+// addTask добавляет новую задачу в базу данных и возвращает ее ID.
+func addTask(task Task) (int, error) {
+	stmt, err := db.Prepare("INSERT INTO scheduler(date, title, comment, repeat) VALUES(?,?,?,?)")
+	if err != nil {
+		return 0, err
```

Comment:
Вот тут так же хорошо везде обернуть 
```suggestion
		return 0, fmt.Errorf("error while prepare query: %w", err)
```